zipline/data/us_equity_minutes.py

```diff
@@ -11,27 +11,402 @@
+        Asset identifier.
+
+    Returns:
+    --------
+    string
+        A path for the bcolz rootdir, including subdirectory prefixes based on
+        the padded string representation of the given sid.
+
+        e.g. 1 is formatted as 00/00/000001.bcolz
+    """
+    padded_sid = format(sid, '06')
+    return os.path.join(
+        # subdir 1 00/XX
+        padded_sid[0:2],
+        # subdir 2 XX/00
+        padded_sid[2:4],
+        "{0}.bcolz".format(str(padded_sid))
+    )
+
+
+class BcolzMinuteBarWriter(object):
+    """
+    Class capable of writing minute OHLCV data to disk into bcolz format.
+
+    Writes a bcolz directory for each individual sid, all contained within
+    a root directory which also contains metadata about the entire dataset.
+
+    Each individual asset's data is stored as a bcolz table with a column for
+    each pricing field: (open, high, low, close, volume)
+
+    The open, high, low, and close columns are integers which are 1000 times
+    the quoted price, so that the data can represented and stored as an
+    np.uint32, supporting market prices quoted up to the thousands place.
+
+    volume is a np.uint32 with no mutation of the tens place.
+
+    The 'index' for each individual asset are a repeating period of minutes of
+    length `minutes_per_day` starting from each market open.
+    The file format does not account for half-days.
+    e.g.:
+    2016-01-19 14:31
+    2016-01-19 14:32
+    ...
+    2016-01-19 20:59
+    2016-01-19 21:00
+    2016-01-20 14:31
+    2016-01-20 14:32
+    ...
+    2016-01-20 20:59
+    2016-01-20 21:00
+
+    All assets are written with a common 'index', sharing a common first
+    trading day. Assets that do not begin trading until after the first trading
+    day will have zeros for all pricing data up and until data is traded.
+
+    'index' is in quotations, because bcolz does not provide an index. The
+    format allows index-like behavior by writing each minute's data into the
+    corresponding position of the enumeration of the aforementioned datetime
+    index.
+
+    The datetimes which correspond to each position are written in the metadata
+    as integer nanoseconds since the epoch into the `minute_index` key.
+    """
+    def __init__(self,
+                 first_trading_day,
+                 rootdir,
+                 market_opens,
+                 minutes_per_day=MINUTES_PER_DAY,
+                 ohlc_ratio=OHLC_RATIO,
+                 expectedlen=DEFAULT_EXPECTEDLEN):
+        """
+        Parameters:
+        -----------
+
+        first_trading_day : datetime-like
+            The first trading day in the data set.
+        
+        rootdir : string
+            Path to the root directory into which to write the metadata and
+            bcolz subdirectories.
+
+        market_opens : pd.Series
+            The market opens used as a starting point for each periodic span of
+            minutes in the index.
+
+        minutes_per_day : int
```

Comment:
I think it would benefit usage of BcolzEquityMinuteWriter to not need to pass the minutes per day on every instantiation.

What do you think of turning this class into `BcolzMinuteWriter` and then subclassing a `BcolzUSEquitiesMinuteWriter` which provides the minutes per day as a default?